src/ci/scripts/free-disk-space-linux.sh

```diff
@@ -247,12 +247,71 @@ cleanSwap() {
     free -h
 }
 
+# Try to find a different drive to put our data on so we don't need to run cleanup.
+# The availability of the disks we're probing isn't guaranteed,
+# so this is opportunistic.
+checkAlternative() {
+    local gha_alt_disk="/mnt"
+
+    # we need ~50GB of space
+    local space_target_kb=$((50 * 1024 * 1024))
+    local available_space_kb=$(df -k "$gha_alt_disk" --output=avail | tail -n 1)
+
+    # mount options that trade durability for performance
+    # ignore-tidy-linelength
+    local mntopts="defaults,discard,journal_async_commit,barrier=0,noauto_da_alloc,lazytime,data=writeback"
+
+    # GHA has a 2nd disk mounted at /mnt that is almost empty.
+    # Check if it's a valid mountpoint and it has enough available space.
+    if mountpoint "$gha_alt_disk" && [ "$available_space_kb" -ge "$space_target_kb" ]; then
+        local blkdev=$(df -k "$gha_alt_disk" --output=source | tail -n 1)
+        echo "Sufficient space available on $blkdev mounted at $gha_alt_disk"
+        # see cleanSwap(), swapfile may be mounted under /mnt
+        sudo swapoff -a || true
+
+        # unmount from original location
+        sudo umount "$gha_alt_disk"
+
+        # remount under the obj dir which is used by docker scripts to write most
+        # of our build output. And apply optimized mount options while we're at it.
+        mkdir ./obj
+        sudo mount $blkdev ./obj -o $mntopts || (sudo dmesg | tail -n 20 ; exit 1)
```

Comment:
added an exit 1 here

Reply:
sorry if I took so long reviewing this.

So if the mount fails, CI fails? Shouldn't we fallback to disk space cleanup?

Reply:
hrrm maybe. Something going wrong here is kind of unexpected. It means the mountpoint does exist but the mount options are wrong, e.g. because it's not ext4.
I suppose microsoft could change their default filesystem choice for some reason...

I guess I'll just print a warning and then fallback.